src/main/java/xbot/common/controls/actuators/XCANMotorController.java

```diff
@@ -223,33 +235,212 @@ public void periodic() {
+        if (distancePerAngleScaleFactor == null) {
+            log.warn("Distance per angle not set for motor controller {}", akitName);
+            return Meters.zero();
+        }
+        return getRawPosition().timesConversionFactor(distancePerAngleScaleFactor);
+    }
+
+    /**
+     * Get the position reported by the motor controller.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     * @return The position reported by the motor controller.
+     */
+    public Angle getPosition() {
+        if (angleScaleFactor == null) {
+            return getRawPosition();
+        }
+        return getRawPosition().timesConversionFactor(angleScaleFactor);
+    }
+
+    /**
+     * Override the position of the motor controller.
+     * @param position The new position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
     public void setPosition(Angle position) {
+        if (angleScaleFactor == null) {
+            setRawPosition(position);
+        } else {
+            setRawPosition(position.timesConversionFactor(invertRatio(angleScaleFactor)));
+        }
+    }
+
+    /**
+     * Override the position of the motor controller.
+     * @param position The new position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are ignored.
+     */
+    public abstract void setRawPosition(Angle position);
+
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
+    public void setPositionTarget(Angle position) {
         setPositionTarget(position, MotorPidMode.DutyCycle);
     }
 
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @param mode The PID mode to use when setting the target position.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
     public void setPositionTarget(Angle position, MotorPidMode mode) {
         setPositionTarget(position, mode, 0);
     }
 
-    public abstract void setPositionTarget(Angle position, MotorPidMode mode, int slot);
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @param mode The PID mode to use when setting the target position.
+     * @param slot The PID slot to use when setting the target position.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
+    public void setPositionTarget(Angle position, MotorPidMode mode, int slot) {
+        if (angleScaleFactor == null) {
+            setRawPositionTarget(position, mode, slot);
+        } else {
+            setRawPositionTarget(position.timesConversionFactor(invertRatio(angleScaleFactor)), mode, slot);
```

Comment:
⭐ maybe a nit, but I think it might be better in these cases to pull out the position setting math into it's own bit of logic and then have a shared call to `setRawPositionTarget` after the `if`, that way if the call ever changes someone can't mess up updating one call but not the other by mistake

Reply:
Refactored